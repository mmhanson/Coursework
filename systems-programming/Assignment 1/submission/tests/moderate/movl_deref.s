main:
	movl	$1012, %eax
	movl	$1, %ebx
	movl	$2, %ecx
	movl	$3, %edx
	movl	%ebx, 0(%eax)
	movl	%ecx, 4(%eax)
	movl	%edx, 8(%eax)
	printr	%eax
	movl	0(%eax), %r8d
	movl	4(%eax), %r9d
	movl	8(%eax), %r10d
	printr	%r8d
	printr	%r9d
	printr	%r10d
	movl	%ebx, -4(%eax)
	movl	%ecx, -8(%eax)
	movl	-4(%eax), %r9d
	movl	-8(%eax), %r10d
	printr	%r9d
	printr	%r10d
	ret
