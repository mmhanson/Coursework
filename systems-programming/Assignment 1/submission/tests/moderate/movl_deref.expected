1012 (0x3f4)
1 (0x1)
2 (0x2)
3 (0x3)
1 (0x1)
2 (0x2)
